org.kde.CrowTranslate.yaml

```diff
@@ -0,0 +1,87 @@
+id: org.kde.CrowTranslate
+sdk: org.kde.Sdk
+runtime: org.kde.Platform
+runtime-version: 5.15-23.08
+add-extensions:
+  org.kde.CrowTranslate.tessdata:
+    directory: share/tessdata
+    version: stable
+    versions: stable;beta;test
```

Comment:
but then there is no such package. not even in the submission queue.

Reply:
Yes, I copied it from [this](https://github.com/flathub/io.github.manisandro.gImageReader/blob/master/io.github.manisandro.gImageReader.yml) app.

> but then there is no such package. not even in the submission queue.

Ah, I thought that it creates a extension for the app. Do I need to package tessdata separately? I don't see ` io.github.manisandro.gImageReader.tessdata` anywhere as well...

I'm new to flatpak.

Reply:
two wrongs don't make a right.

Reply:
Let's try to figure out how to do properly. I'm not familiar with this so we'll have to search in the docs.

Reply:
`man flatpak-metadata`

[...]

```
       version (string)
           The branch to use when looking for the extension. If this is not specified, it defaults to the branch of the application
           or runtime that the extension point is for. Available since 0.4.1.

       versions (string)
           The branches to use when looking for the extension. If this is not specified, it defaults to the branch of the
           application or runtime that the extension point is for. Available since 0.9.1, and backported to the 0.8.x branch in
           0.8.4.

```

Reply:
What it doesn't say is that they are equivalent one being the plural form.

That doesn't make an extension being built. That just declare the extension point.

`man flatpak-manifest`

```
       bundle (boolean)
           If this is true, then the data created in the extension directory is omitted from the result, and instead packaged in a
           separate extension.

```

Reply:
> Let's try to figure out how to do properly

Reading it right now. Looks like I need to create a separate manifest for it and add a special entry to my app metadata.
We probably need to notify the maintainer of `io.github.manisandro.gImageReader` as well about it.

> man flatpak-metadata

We aren't talking about the version string, but about the whole `add-extensions` entry.

Reply:
> Reading it right now. Looks like I need to create a separate manifest for it and add a special entry to my app metadata.

no.

Reply:
> That doesn't make an extension being built. That just declare the extension point.

It's because `bundle` is false by default?
So instead of setting a version and creating a separate manifest, I can replace `version` with `bundle: true`?

Reply:
> Yes, I copied it from [this](https://github.com/flathub/io.github.manisandro.gImageReader/blob/master/io.github.manisandro.gImageReader.yml) app.

See the tip here https://docs.flatpak.org/en/latest/extension.html#extension-point Since you don't have a beta application you don't need this. `test` is not needed, it's a temporary repo.

> It's because bundle is false by default?

Yes

https://docs.flatpak.org/en/latest/extension.html#bundled-extensions

Reply:
I think I figured it out. I enabled bundling and now the extension creates properly. I also set version that matches the tessdata version.

Opened an issue no notify the maintainer of io.github.manisandro.gImageReader about it: https://github.com/flathub/io.github.manisandro.gImageReader/issues/70

@hfiguiere Thanks for pointing it out, but if your goal is to help the community, try to give helpful advice to newcomers instead of just saying "it's wrong".

Reply:
That's what I thought, thanks!
It looks like, in this case, the tessada was installed as part of the main manifest, but the extension relied on a separate, non-existent manifest. The author has just applied the same fix: https://github.com/flathub/io.github.manisandro.gImageReader/pull/71